src/boot_O2/boot_80086760.c

```diff
@@ -1,27 +1,211 @@
+    return poly;
+}
+
+/**
+ * Ditto for two quadrants, the rest of the range.
+ * Uses the formulae arctan(x) = pi/2 - arctan(1/x)
+ * and arctan(x) = pi/4 - arctan( (1-x)/(1+x) )
+ * to extend the range in which the series computed by func_80086834 is a good approximation
+ */
+// Math_FAtanTaylorF
+f32 func_80086880(f32 x) {
+    f32 t;
+    f32 q;
+
+    if (x > 0.0f) {
+        t = x;
+    } else if (x < 0.0f) {
+        t = -x;
+    } else if (x == 0.0f) {
+        return 0.0f;
+    } else {
+        return qNaN0x10000;
+    }
+
+    if (t <= M_SQRT2 - 1.0f) {
+        return func_80086834(x);
+    }
+
+    if (t >= M_SQRT2 + 1.0f) {
+        q = M_PI / 2 - func_80086834(1.0f / t);
+    } else { // in the interval (\sqrt{2} - 1, \sqrt{2} + 1)
+        q = M_PI / 4 - func_80086834((1.0f - t) / (1.0f + t));
+    }
+
+    if (x > 0.0f) {
+        return q;
+    } else {
+        return -q;
+    }
+}
+
+/**
+ * Arctangent approximation using a continued fraction
+ * Cf. https://en.wikipedia.org/wiki/Gauss%27s_continued_fraction#The_series_2F1_2 ,
+ * https://dlmf.nist.gov/4.25#E4
+ */
+// Math_FAtanContFracF
+f32 func_800869A4(f32 x) {
+    s32 sector;
+    f32 z;
+    f32 conv;
+    f32 sq;
+    s32 i;
+
+    if (x >= -1.0f && x <= 1.0f) {
+        sector = 0;
+    } else if (x > 1.0f) {
+        sector = 1;
+        x = 1.0f / x;
+    } else if (x < -1.0f) {
+        sector = -1;
+        x = 1.0f / x;
+    } else {
+        return qNaN0x10000;
+    }
+
+    // Builds the continued fraction from the innermost fraction out
+    sq = SQ(x);
+    conv = 0.0f;
+    z = 8.0f;
+    for (i = 8; i != 0; i--) {
+        conv = SQ(z) * sq / (2.0f * z + 1.0f + conv);
+        z -= 1.0f;
+    }
+    conv = x / (1.0f + conv);
+
+    if (sector == 0) {
+        return conv;
+    } else if (sector > 0) {
+        return M_PI / 2 - conv;
+    } else {
+        return -M_PI / 2 - conv;
+    }
+}
+
+// Math_FAtanF
+/**
+ * Single-argument arctangent, only used by the two-argument function.
+ * Nothing else sets the bss variable gUseAtanContFrac, so the Maclaurin series is always used
+ */
+f32 func_80086AF0(f32 x) {
+    if (!gUseAtanContFrac) {
+        return func_80086880(x);
+    } else {
+        return func_800869A4(x);
+    }
+}
+
+// Math_FAtan2F
+/**
+ * Main two-argument arctangent function
+ */
+f32 func_80086B30(f32 y, f32 x) {
+    if (x == 0.0f) {
+        if (y == 0.0f) {
+            return 0.0f;
+        } else if (y > 0.0f) {
+            return M_PI / 2;
+        } else if (y < 0.0f) {
+            return -M_PI / 2;
+        } else {
+            return qNaN0x10000;
+        }
+    } else if (x >= 0.0f) {
+        return func_80086AF0(y / x);
+    } else if (y < 0.0f) {
+        return func_80086AF0(y / x) - M_PI;
+    } else {
+        return M_PI - func_80086AF0(-(y / x));
+    }
+}
+
+// Math_FAsinF
+f32 func_80086C18(f32 x) {
+    return func_80086B30(x, sqrtf(1.0f - SQ(x)));
```

Comment:
```suggestion
    return func_80086B30(x, sqrtf(SQ(1.0f) - SQ(x)));
```
This might be trivial and not needed, but it's consistent so you can choose to include it or not

Reply:
I'm not sure that will even match, but I'll check.

Reply:
It does match, but on reflection I don't really like it. You'd never write it in practice anyway.